Sources/CodeEditTextView/TreeSitter/TreeSitterClient+LanguageLayer.swift

```diff
@@ -0,0 +1,52 @@
+//
+//  TreeSitterClient+LanguageLayer.swift
+//  
+//
+//  Created by Khan Winter on 3/8/23.
+//
+
+import Foundation
+import CodeEditLanguages
+import SwiftTreeSitter
+
+extension TreeSitterClient {
+    class LanguageLayer: Hashable {
+        /// Initialize a language layer
+        /// - Parameters:
+        ///   - id: The ID of the layer.
+        ///   - parser: A parser to use for the layer.
+        ///   - supportsInjections: Set to true when the langauge supports the `injections` query.
+        ///   - tree: The tree-sitter tree generated while editing/parsing a document.
+        ///   - languageQuery: The language query used for fetching the associated `queries.scm` file
+        ///   - ranges: All ranges this layer acts on. Must be kept in order and w/o overlap.
+        init(id: TreeSitterLanguage,
+             parser: Parser,
+             supportsInjections: Bool,
+             tree: Tree? = nil,
+             languageQuery: Query? = nil,
+             ranges: [NSRange]) {
```

Comment:
### [CodeStyle]:

Maybe try to not indent to much. In the `CodeEdit` repo this is not allowed by SwiftLint. We should keep it consistent.

```suggestion
        init(
            id: TreeSitterLanguage,
            parser: Parser,
            supportsInjections: Bool,
            tree: Tree? = nil,
            languageQuery: Query? = nil,
            ranges: [NSRange]
        ) {
```